src/SoupRot.ahk

```diff
@@ -0,0 +1,159 @@
+﻿/* Written by Masonjar13
+
+	A static-class for a length-based rotational cipher.
+	
+	Works for any unicode string by first encoding the string
+	to base64. Decodes back to UTF-8 (could be altered to support
+	binary data).
+
+	Methods & Parameters:
+---------------
+	soupRot.enc()	- encode string
+		str			- string to encode
+		mult 		- rotation iteration count
+		junk		- random character count to be added (default: 0)
+	
+	soupRot.dec()	- decode string
+		str			- string to decode
+		mult		- rotation iteration count used to encode
+		junk		- random character count used to encode (default: 0)
+	
+	dependencies (can be found at https://github.com/Masonjar13/AHK-Library)
+		randStr()
+		rand()
+		ifContains()
+		isDigit()
+		b64e()		- encode string
+		b64d()		- decode string (utf-8)
+		strPutVar()	- for b64e()
+---------------
+
+	Example:
+------------
+iterationCnt:=a_tickCount
+junk:=0
+str:="Hello° µWorld"
+
+encStr:=soupRot.enc(str,iterationCnt,junk)
+decStr:=soupRot.dec(encStr,iterationCnt,junk)
+
+msgbox,,SoupRot,% encStr "`n`n" decStr
+------------
+
+*/
+
+class soupRot {

```

Comment:
Neat! first implementation that uses a `class`. All other implementations are single functions for simple in/out or enc/dec functions. That said, I don't have anything against this. I believe there's also an issue/work-item to make libcrypt into a class. For now, please prefix it with `LC_` for it to comply with all other functions, likely: `LC_soupRot`... to make it more "self-contained" or rather reduce loose names in global space -> having everything prefixed with `LC_`.

Reply:
Those dependencies were the reason for the class, actually. I work very modularly and rely heavily on my stdlib. It would definitely be better to have this entire library as a class (I certainly can work on that later), but I'll see what I can do about reducing the functions.